[GLib] Stop setting AuxiliaryProcessProxy::processID until it can be done correctly

mcatanzaro opened the pull request:
#### 0b759034fb5fa72b32787132ad54b846b1372c74
<pre>
[GLib] Stop setting AuxiliaryProcessProxy::processID until it can be done correctly
<a href="https://bugs.webkit.org/show_bug.cgi?id=266489">https://bugs.webkit.org/show_bug.cgi?id=266489</a>

Reviewed by NOBODY (OOPS!).

We currently set AuxiliaryProcessProxy::processID to non-useful values,
causing weird behavior like suspending/resuming the flatpak-spawn or
bwrap intermediate processes instead of the intended WebKit auxiliary
process, terminating the intermediate process instead of the WebKit
process, etc. Let&apos;s temporarily disable all pid-based functionality
until we find a way to fix the pids we use.

This will break the FocusedFrameAfterCrash test, which manually kills
the child process using the pid that&apos;s no longer set.

* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):
* Tools/TestWebKitAPI/glib/TestExpectations.json:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0b759034fb5fa72b32787132ad54b846b1372c74

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52827 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32164 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5314 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56106 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3550 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55132 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38889 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3275 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42877 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2296 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54925 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29889 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45604 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23992 "Found 1 new API test failure: /TestWebKit:WebKit.FocusedFrameAfterCrash (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26972 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2921 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1709 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48845 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3072 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57699 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27968 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3055 "Found 1 new test failure: imported/w3c/web-platform-tests/html/browsers/the-window-object/open-close/open-features-tokenization-noreferrer.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50275 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29188 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45796 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49557 "Found 1 new API test failure: /TestWebKit:WebKit.FocusedFrameAfterCrash (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30107 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28942 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
```diff
@@ -232,12 +232,31 @@ void ProcessLauncher::launchProcess()
+    // process in the pid namespace of the UI process, but this is probably
+    // impossible to do correctly for Flatpak. Until we're able to fix this, set
+    // it to 0 always to ensure that Flatpak users do not get worse behavior
+    // than non-Flatpak users. This will prevent us from accidentally adding
+    // more functionality that works for developers but doesn't work for users.
```
I'm not sure I understand this, are you assuming that all users use flatpak and developers don't? I see this in a different way, why do we want to break non-flatpak users because of flatpak specific issue?

mcatanzaro replied:
It's already broken for non-flatpak users too; the m_processID refers to the bwrap process and not the web process as intended, so e.g. we suspend only the manager process while leaving the real child process still running.

It's probably easy to fix that for bubblewrap sandbox, but I haven't tried.

aperezdc replied:
I went to check and I am disappointed that `bwrap` does not have an option to forward signals to child processes 😢

mcatanzaro replied:
We could add one easily enough, but it wouldn't work here anyway, because the signal we try to send is SIGSTOP. Sending the signal would inherently stop bwrap from forwarding it. :)

carlosgcampos commented on Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
```diff
@@ -232,12 +232,31 @@ void ProcessLauncher::launchProcess()
+    // it to 0 always to ensure that Flatpak users do not get worse behavior
+    // than non-Flatpak users. This will prevent us from accidentally adding
+    // more functionality that works for developers but doesn't work for users.
+    // m_processID is optional and there should (hopefully) be no code in WebKit
+    // that depends on it without checking whether it is 0.
```
There's one api test failing in both gtk and wpe EWS.

mcatanzaro committed:
[GLib] Stop setting AuxiliaryProcessProxy::processID until it can be done correctly
https://bugs.webkit.org/show_bug.cgi?id=266489

Reviewed by NOBODY (OOPS!).

We currently set AuxiliaryProcessProxy::processID to non-useful values,
causing weird behavior like suspending/resuming the flatpak-spawn or
bwrap intermediate processes instead of the intended WebKit auxiliary
process, terminating the intermediate process instead of the WebKit
process, etc. Let's temporarily disable all pid-based functionality
until we find a way to fix the pids we use.

This will break the FocusedFrameAfterCrash test, which manually kills
the child process using the pid that's no longer set.

* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):
* Tools/TestWebKitAPI/glib/TestExpectations.json:
